crates/ruff_python_parser/src/error.rs

```diff
@@ -449,6 +449,7 @@ pub enum UnsupportedSyntaxErrorKind {
     Match,
     Walrus,
     ExceptStar,
+    TypeStmt,
```

Comment:
Should we use type alias instead of type statement? I would find this more approachable as a user, unless the term is ambiguous. 

```suggestion
    TypeAlias,
```

https://docs.python.org/3/library/typing.html#type-aliases

If not `TypeAlias`, then rename to `TypeStatement`

Reply:
pyright calls them "type alias statements," which I think sounds good for the user-facing message. For the variant name, I don't have strong feelings either way. They do call it a [`type_stmt`](https://docs.python.org/3/reference/simple_stmts.html#the-type-statement) in the Python reference, so I might lean toward `TypeStatement` if you prefer that to `Stmt`.

Reply:
I'll just call it `TypeAliasStatement`, I see that's what we call the parser function now.